e2e_testing/torchscript/basic.py

```diff
@@ -1179,3 +1179,56 @@ def forward(self, a):
 @register_test_case(module_factory=lambda: BoolTensorReturnMixedModule())
 def BoolTensorReturnMixedModule_basic(module, tu: TestUtils):
     module.forward(torch.tensor([[1, 0], [0,1]], dtype=torch.bool))
+
+# ==============================================================================
+class TModuleRank2(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([-1, -1], torch.float32, True),
+    ])
+    def forward(self, lhs):
+        return torch.t(lhs)
+
+
+@register_test_case(module_factory=lambda: TModuleRank2())
+def TModuleRank2_basic(module, tu: TestUtils):
+    module.forward(tu.rand(3, 4))
+
+class TModuleRank1(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([-1], torch.float32, True),
+    ])
+    def forward(self, lhs):
+        return torch.t(lhs)
+
+
+@register_test_case(module_factory=lambda: TModuleRank1())
+def TModuleRank1_basic(module, tu: TestUtils):
+    module.forward(tu.rand(3))
+
+class TModuleRank0(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([-1], torch.float32, True),
+    ])
+    def forward(self, lhs):
+        return torch.t(lhs)
+
+
+@register_test_case(module_factory=lambda: TModuleRank0())
+def TModuleRank0_basic(module, tu: TestUtils):
+    module.forward(tu.rand(1))
+
```

Comment:
Any idea why TModuleRank1_basic and TModuleRank0_basic pass unexpectedly?

Reply:
New tests are set as "expected to fail" by default on the TOSA side. To add it to the "expected to pass" set, add your test to this list

https://github.com/sjarus/torch-mlir/blob/4bb8668f8db1fd2829ba9734381c7ecbf540eabe/e2e_testing/torchscript/xfail_sets.py#L24

The reason those two are passing on the TOSA side despite TOSA not yet having support for the transpose op is because for those two cases, the transpose op gets removed during decomposition, which happens before the `TorchToTosa` pass takes place.